overlord/devicestate/handlers.go

```diff
@@ -63,13 +68,54 @@ func deviceAPIBaseURL() string {
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength        uint64 = 4096
+	retryInterval           = 60 * time.Second
+	deviceAPIBase           = deviceAPIBaseURL()
+	requestIDURL            = deviceAPIBase + "request-id"
+	serialRequestURL        = deviceAPIBase + "devices"
+	sshKeyFile              = "/tmp/snapd.key.tmp"
```

Comment:
we need to think where to best put these, though we know ssh-keygen will make sure they are root only accessible